src/scripts/generate_covid_case_datamart.py

```diff
@@ -0,0 +1,111 @@
+import pathlib
+import json
+from typing import List, Callable
+from utils import read_blob, record_combination_func, write_blob
+
+
+def standardize_lab_result(result: str) -> str:
+    """
+    Given a covid test result and dictionary whose keys (POSITIVE, NEGATIVE, and
+    INCONCLUSIVE) represent standard results and whose values are lists containing
+    non-standard results that map to the associated standard result then return a
+    standardized result. Return null is no standard result could not be found.
+    """
+    standard = json.loads(
+        open(
+            pathlib.Path(__file__).parent / "assets" / "covid_lab_results_standard.json"
+        ).read()
+    )
+    result = result.upper()
+    for standard_result, non_standard_results in standard.items():
+        if result in non_standard_results:
+            return standard_result
+
+    return ""
+
+
+def identify_covid_cases(
+    patient_labs: pd.DataFrame, agg_function: Callable
+) -> List[pd.Series]:
+    """
+    Given a data frame containing a single patient's COVID labs and an aggregation
+    function, review the labs for possible COVID cases and return a list containing a
+    for each identified case. Data for fields specific to the patient and not the
+    infection result from aggregating accross all labs.
+
+    **** WARNING ****
+
+    As currently written this function applies an approximation of the official CDC/CTSE
+    COIVD case defintion. In its current form this function is intended as a proof of
+    concept for development purposes.
+
+    Known Limitations:
+    1.  Suspect cases resulting from antibody tests are not considered.
+    2.  No check has been implemented for a negative PCR within 48 hours of a positive
+        Antigen that would negate the case.
+    3.  No consideration has been given to how the case definition has changed over the
+        course of the pandemic.
+    """
+    case_list = []
+    patient_labs.sort_values(by=["effectiveDateTime"], ascending=True)
+    positive_labs = patient_labs.loc[patient_labs.result == "POSITIVE"].reset_index()
+    remaining_labs = len(positive_labs)
+    while remaining_labs > 0:
+        collection_date = positive_labs.effectiveDateTime[0]
+        current_infection_labs = positive_labs.loc[
+            positive_labs.effectiveDateTime.map(lambda x: (x - collection_date).days)
+            <= 90
+        ]
+        if "pcr" in current_infection_labs.type.to_list():
+            status = "confirmed"
+        else:
+            status = "probable"
+        case = patient_labs.agg(record_combination_func)
+        case["effectiveDateTime"] = collection_date
+        case["status"] = status
+        case_list.append(case)
+        positive_labs = positive_labs.loc[
+            positive_labs.effectiveDateTime.map(lambda x: (x - collection_date).days)
+            > 90
+        ].reset_index()
+        remaining_labs = len(positive_labs)
+    return case_list
+
+
+if __name__ == "__main__":
+
```

Comment:
This is just food for thought, should these variables be read from some sort of env variable file ? Is this the best place for these ?

Reply:
Yeah, I have been wondering the same thing. I don't have a sense of how this code might live on beyond 5/3 or what it looks like "in production" beyond just being simple .py files to run locally ad-hoc. If you have thoughts I'm open to them. Happy to refactor into a config file for now if you think it is worth it.

Reply:
Personally I'd just leave them in the .py file for right now. Precisely because we don't know the answers to those (good) questions, I have zero issue with our "analytics" workflow being "some dude running scripts on his laptop" ("some dude" in this case being one of us 🤣 ). If we want to productionize this for more robust analysis at a later date we'll want to make other changes anyway so imo this is fine as is.